contracts/MixinResolver.sol

```diff
@@ -14,6 +14,26 @@ contract MixinResolver is Owned {
 
     uint public constant MAX_ADDRESSES_FROM_RESOLVER = 24;
 
+    bool public isCached;
+
+    // This modifier must be used on all functions that use a cached address to make sure
+    // They always fetch the latest cache
+    modifier validateCache() {
+        if(!isCached) {
```

Comment:
Might be cleaner to call a lazy internal function in the validateCache modifier.